server/controllers/ratingController.js

```diff
@@ -0,0 +1,48 @@
+import ratingStars from 'star-ratings';
+import AccommodationHelper from '../helpers/accommodationHelper';
+import RequestHelper from '../helpers/requestHelper';
+
+/**
+ * This class contains all methods
+ * required to rate
+ *  an accommodation.
+ */
+class RatingController {
+  /**
+   * This method approve the trip request.
+   * @param {object} req The http request.
+   * @param {object} res The response.
+   * @returns {object} The status and some the rating of the accommodation.
+   */
+  static async rateAccommodation(req, res) {
+    const accommodation = await AccommodationHelper.findAccommodation('id', parseInt(req.params.id, 10));
+    const { stars } = req.body;
+    const { ratings } = accommodation;
+
+    if (stars === 1) {
+      ratings[0] += 1;
+    } else if (stars === 2) {
+      ratings[1] += 1;
+    } else if (stars === 3) {
+      ratings[2] += 1;
+    } else if (stars === 4) {
+      ratings[3] += 1;
+    } else {
+      ratings[4] += 1;
+    }
+
+    await AccommodationHelper.updateRatings(ratings, parseInt(req.params.id, 10));
+    await RequestHelper.updateRated(parseInt(req.params.id, 10), req.requesterUser.id);
+    const ratedAccommodation = await AccommodationHelper.findAccommodation('id', parseInt(req.params.id, 10));
```

Comment:
Hi @Nigorjeanluc This is too much for a rating. 4 APIs, It is tooooo much. we should be having only one or two maximum. As you could not do it, Let us do it an other way, a table of ratings and do an association of ratings belonging to accommodation.